json-core/src/main/java/io/micronaut/json/JsonConfig.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.json;
+
+public final class JsonConfig {
```

Comment:
How is this configured? Does it make sense to make it `@ConfigurationProperties` with defaults instead of static state?

Reply:
Alao needs javadoc if it is public API

Reply:
I've renamed this to `JsonStreamConfig` and added docs.

Right now, this config is constructed from the `ObjectMapper`. That means that it comes indirectly from the `JacksonConfiguration`. However, in some cases (e.g. when jackson config is changed through `@JacksonConfiguration`), the settings of the object mapper change and so the properties of this class also change. I don't think it should be made `@ConfigurationProperties` for that reason.